service-map/src/main/java/fi/mml/map/mapwindow/util/OskariLayerWorker.java

```diff
@@ -125,16 +165,35 @@ public static JSONObject getListOfMapLayers(final List<OskariLayer> layers, fina
         final JSONArray layersList = new JSONArray();
         start = System.currentTimeMillis();
         for (OskariLayer layer : layers) {
+        	
             final String permissionKey = layer.getType()+ "+" + layer.getUrl() + "+" + layer.getName();
             if (layer.getParentId() == -1 && !resources.contains(permissionKey)) {
                 // not permitted if resource NOT found in permissions!
                 // sublayers can pass through since their parentId != -1
                 continue;
             }
             try {
+            	String boundingBoxWKT = layer.getCapabilities().optString("geom");
+            	if(boundingBoxWKT != null && boundingBoxWKT.length() > 0) {
+            		Geometry jtsGeometry = null;
+            	    WKTReader reader = new WKTReader();
+            	    jtsGeometry = reader.read(boundingBoxWKT);
+            	    CoordinateReferenceSystem sourceCRS = CRS.decode("EPSG:4326", true);
+            	    String strTargetCRS = crs;
+            	    if(layer.getSrs_name() != null) {
+            	    	strTargetCRS = layer.getSrs_name();
+            	    }
+                    CoordinateReferenceSystem targetCRS = CRS.decode(strTargetCRS, true);
+                    MathTransform transform = CRS.findMathTransform(sourceCRS, targetCRS, true);
+                    Geometry targetGeometry = JTS.transform(jtsGeometry, transform);
+                    layer.setGeometry(targetGeometry.toText());
+            	}
                 final JSONObject layerJson = FORMATTER.getJSON(layer, lang, isSecure);
                 //log.debug("Generated JSON");
                 if (layerJson != null) {
+                    if(layer.getGeometry() != null && layer.getGeometry().length() > 0) {
+                    	JSONHelper.putValue(layerJson, "geom", layer.getGeometry());
```

Comment:
Instead of this if-statement call transformWKTGeom() here. Remove call to transform from GetMapLayersHandler so it's not done twice: https://github.com/oskariorg/oskari-server/blob/develop/control-base/src/main/java/fi/nls/oskari/control/layer/GetMapLayersHandler.java#L33-L36

Reply:
Actually call from GetMapLayersHandler is still required as crs is not passed from there. Do we still need GetMapLayersHandler with this PR? Maybe it can be removed?